src/codegen/compile.ml

```diff
@@ -3675,29 +3675,40 @@ module Blob = struct
 
     get_blob
 
-  (* Lexicographic blob comparison. Expects two blobs on the stack *)
+  (* Lexicographic blob comparison. Expects two blobs on the stack.
+     Either specialized to a specific comparision operator, and returns a boolean,
+     or implements the generic comparison, returning -1, 0 or 1 as Int8.
+  *)
   let rec compare env op =
+    let word32_cmp get_a get_b =
+        (* Wasm has no compare instruction for Word32 that returns -1, 0, 1. Here is one *)
+        get_a ^^ get_b ^^ G.i (Compare (Wasm.Values.I32 I32Op.LtU)) ^^
+        G.if0 (compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 (-1)) ^^ G.i Return) G.nop ^^
+        get_a ^^ get_b ^^ G.i (Compare (Wasm.Values.I32 I32Op.GtU)) ^^
+        G.if0 (compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 1) ^^ G.i Return) G.nop ^^
+        compile_unboxed_const (TaggedSmallWord.vanilla_lit Type.Int8 0) in
```

Comment:
Is it really necessary to leave a 0 on the stack? The other two cases just exit the function, staying in it means the comparison gave an equality. (I am assuming this is being in a loop, so it seems useless stack traffic.)

Reply:
This should be more straightforward